acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/ThreeLeggedOAuthAcceptanceTest.java

```diff
@@ -0,0 +1,209 @@
+public class ThreeLeggedOAuthAcceptanceTest extends AbstractJUnitTest {
+
+    @Inject
+    private JenkinsController controller;
+
+    private JenkinsOAuthClient oAuthClient;
+
+    private FreeStyleJob job;
+
+    private User admin;
+    private User user1;
+    private User user2;
+
+    @Before
+    public void setup() throws Exception {
+        JenkinsApplinksClient applinksClient = new JenkinsApplinksClient(getBaseUrl());
+        OAuthConsumer oAuthConsumer = applinksClient.createOAuthConsumer();
+        oAuthClient = new JenkinsOAuthClient(getBaseUrl(), oAuthConsumer.getKey(), oAuthConsumer.getSecret());
+
+        // Enable security
+        GlobalSecurityConfig securityConfig = new GlobalSecurityConfig(jenkins);
+        securityConfig.configure();
+        JenkinsDatabaseSecurityRealm realm = securityConfig.useRealm(JenkinsDatabaseSecurityRealm.class);
+        realm.allowUsersToSignUp(true);
+        securityConfig.save();
+
+        user1 = realm.signup("stash-user" + randomNumeric(8));
+        user2 = realm.signup("stash-user" + randomNumeric(8));
+        admin = realm.signup("admin-user" + randomNumeric(8));
+
+        securityConfig.configure();
+        ProjectBasedMatrixAuthorizationStrategy matrixAuthzConfig =
+                authorizeUserAsAdmin(admin.fullName(), securityConfig)
+                        .useAuthorizationStrategy(ProjectBasedMatrixAuthorizationStrategy.class);
+        MatrixRow user1MatrixRow = matrixAuthzConfig.addUser(user1.fullName());
+        user1MatrixRow.on(OVERALL_READ);
+        MatrixRow user2MatrixRow = matrixAuthzConfig.addUser(user2.fullName());
+        user2MatrixRow.on(OVERALL_READ);
+        securityConfig.save();
+
+        job = jenkins.jobs.create();
+        job.save();
+
+        job.configure();
+        ProjectMatrixProperty pmp = new ProjectMatrixProperty(job);
+        pmp.enable.check();
+        MatrixRow user1JobMatrixRow = pmp.addUser(user1.fullName());
+        user1JobMatrixRow.on(ITEM_READ);
+        MatrixRow user2JobMatrixRow = pmp.addUser(user2.fullName());
+        user2JobMatrixRow.on(ITEM_BUILD, ITEM_READ);
+        job.save();
+
+        jenkins.logout();
+    }
+
+    @Test
+    public void testAuthorize() {
+        OAuth1AccessToken user1AccessToken = getAccessToken(user1);
+        OAuth1AccessToken user2AccessToken = getAccessToken(user2);
+
+        String jobBuildPostUrl = String.format("%s/job/%s/build", removeEnd(getBaseUrl(), "/"), job.name);
+        OAuthRequest buildRequest = new OAuthRequest(Verb.POST, jobBuildPostUrl);
+        buildRequest.addHeader("Accept", "application/json");
+
+        Response user2BuildResponse = oAuthClient.execute(buildRequest, user2AccessToken);
+        assertThat(user2BuildResponse, successful());
+
+        Response user1BuildResponse = oAuthClient.execute(buildRequest, user1AccessToken);
+        assertThat(user1BuildResponse, unauthorized());
+    }
+
+    private OAuth1AccessToken getAccessToken(User user) {
```

Comment:
Unfortunately, github isn't treating this as a rename/edit, but inside this method is the same as the OAuth dance that was in `Applinks3LOAuthTest` before.